### PR TITLE
Add instructions to install Miniconda

### PR DESCRIPTION
**Pull request by seisman**

### CODE REVIEW

File: source/programming/python.md [hunk 1]
```diff
@@ -1,16 +1,88 @@
...
+
+## 简介
+
+[Python](https://www.python.org/) 是一种广泛使用的通用编程语言，其具有语法简单、
+功能强大等优点，是目前地震学科研最常用的编程语言之一。
+
+## 安装 Miniconda
+
+Python 是一门解释型语言，因而需要专门的解释器去执行 Python 指令。
+尽管 Linux/macOS 系统已经内置了 Python 解释器，但是建议用户不要使用系统内置它，
```
Comment (core-man) [inline source/programming/python.md:17]: I cannot understand this sentence.
```
但是建议用户不要使用系统内置它，
```

File: source/programming/python.md [hunk 2]
```diff
@@ -1,16 +1,88 @@
...
+以免误操作破坏系统内置 Python 造成系统出现问题。
+**建议用户安装 Miniconda 并使用其提供的 `conda` 命令管理和安装 Python 及其模块。**
+
+:::{dropdown} Python vs Anaconda vs Miniconda
+:color: info
+:icon: info
+
+在安装和使用 Miniconda 前，读者有必要了解 Python、Anaconda 和 Miniconda 三者
+之间的区别于联系：
```
Comment (core-man) [inline source/programming/python.md:26]:
```suggestion
之间的区别与联系：
```

File: source/programming/python.md [hunk 3]
```diff
@@ -1,16 +1,88 @@
...
+
+Python 解释器
+: 从 [Python 官方网站下载](https://www.python.org/downloads/)的 Python 安装包
```
Comment (core-man) [inline source/programming/python.md:29]: Maybe we don't need to include "下载" in the hyperlink, no?
```suggestion
: 从 [Python 官方网站](https://www.python.org/downloads/) 下载的 Python 安装包
```

File: source/programming/python.md [hunk 4]
```diff
@@ -1,16 +1,88 @@
...
+  就提供了一个 Python 解释器，其包含了 Python 的核心模块和库，是运行 Python
```
Comment (core-man) [inline source/programming/python.md:30]:
```suggestion
  只提供了一个 Python 解释器，仅包含 Python 的核心模块和库，是运行 Python
```

File: source/programming/python.md [hunk 5]
```diff
@@ -1,16 +1,88 @@
...
+  指令/脚本所必须的软件。
```
Comment (core-man) [inline source/programming/python.md:31]: I think we could make the last sentence much simpler.
```suggestion
  脚本所必需的。
```

File: source/programming/python.md [hunk 6]
```diff
@@ -1,16 +1,88 @@
...
+
+Anaconda
+: [Anaconda](https://www.anaconda.com/) 是一个 Python **发行版**，其不仅提供了
```
Comment (core-man) [inline source/programming/python.md:34]:
```suggestion
: [Anaconda](https://www.anaconda.com/) 是一个 Python **发行版**，不仅提供了
```

File: source/programming/python.md [hunk 7]
```diff
@@ -1,16 +1,88 @@
...
+  Python 解释器，还内置了很多 Python 开发工具，以及众多科学计算相关的库，形成了
```
Comment (core-man) [inline source/programming/python.md:35]:
```suggestion
  Python 解释器，还内置了很多 Python 开发工具与众多科学计算相关的库，形成了
```

File: source/programming/python.md [hunk 8]
```diff
@@ -1,16 +1,88 @@
...
+  一个可以开箱即用的 Python 科学计算环境，省去了自行配置科学计算环境的麻烦。
+  同时，Anaconda 还提供了强大的软件包管理工具 `conda`，可以方便地安装模块和管理环境。
```
Comment (core-man) [inline source/programming/python.md:37]:
```suggestion
  Anaconda 还提供了强大的软件包管理工具 `conda`，可以方便地安装模块和管理环境。
```

File: source/programming/python.md [hunk 9]
```diff
@@ -1,16 +1,88 @@
...
+  - 安装了很多平时用不到的模块，进而导致安装新模块时会可能出现版本冲突
+
+Miniconda
+: [Miniconda](https://docs.conda.io/en/latest/miniconda.html) 是 Anaconda 的精简版。
+  它继承了 Anaconda 的优点，同时避免了 Anaconda 的臃肿。其安装包只有约 50 MB，安装
+  通常也只需要数十秒。安装 Miniconda 相当于安装了 Python 解释器 + 包管理器 `conda`。
 :::
 
-## Python 简介
+下面展示了如何 在 Linux 系统下安装 Miniconda：
```
Comment (core-man) [inline source/programming/python.md:52]:
```suggestion
下面展示了如何在 Linux 系统下安装 Miniconda：
```

File: source/programming/python.md [hunk 10]
```diff
@@ -1,16 +1,88 @@
...
+  指令/脚本所必须的软件。
+
+Anaconda
+: [Anaconda](https://www.anaconda.com/) 是一个 Python **发行版**，其不仅提供了
+  Python 解释器，还内置了很多 Python 开发工具，以及众多科学计算相关的库，形成了
+  一个可以开箱即用的 Python 科学计算环境，省去了自行配置科学计算环境的麻烦。
+  同时，Anaconda 还提供了强大的软件包管理工具 `conda`，可以方便地安装模块和管理环境。
+  安装 Anaconda 相当于安装了 Python 解释器 + 数百个科学计算相关模块 + 包管理器 `conda`。
+
+  尽管 Anaconda 有很多优点，其也有很明显的缺点：
```
Comment (core-man) [inline source/programming/python.md:40]:
```suggestion
  尽管 Anaconda 有很多优点，其也有明显的缺点：
```

File: source/programming/python.md [hunk 11]
```diff
@@ -1,16 +1,88 @@
...
+  - 安装包非常大（超过 500 MB）
+  - 安装过程耗时长（一般超过 5 分钟）
+  - 安装后占用大量硬盘空间（一般超过 3 GB）
+  - 安装了很多平时用不到的模块，进而导致安装新模块时会可能出现版本冲突
+
+Miniconda
+: [Miniconda](https://docs.conda.io/en/latest/miniconda.html) 是 Anaconda 的精简版。
+  它继承了 Anaconda 的优点，同时避免了 Anaconda 的臃肿。其安装包只有约 50 MB，安装
+  通常也只需要数十秒。安装 Miniconda 相当于安装了 Python 解释器 + 包管理器 `conda`。
```
Comment (core-man) [inline source/programming/python.md:49]:
```suggestion
  通常也只需要数十秒。安装 Miniconda 相当于安装了 Python 解释器 + 核心模块/库 + 包管理器 `conda`。
```

File: source/programming/python.md (hunk 1 above)
Comment (seisman) [inline source/programming/python.md:17]:
```suggestion
尽管 Linux/macOS 系统已经内置了 Python 解释器，但是建议用户不要使用它，
```

File: source/programming/python.md [hunk 12]
```diff
@@ -1,16 +1,88 @@
...
+---
+
+## 简介
+
+[Python](https://www.python.org/) 是一种广泛使用的通用编程语言，其具有语法简单、
+功能强大等优点，是目前地震学科研最常用的编程语言之一。
+
+## 安装 Miniconda
+
+Python 是一门解释型语言，因而需要专门的解释器去执行 Python 指令。
```
Comment (core-man) [inline source/programming/python.md:16]:
```suggestion
Python 是一种解释型语言，需要专门的解释器去执行 Python 代码。
```

File: source/programming/python.md [hunk 13]
```diff
@@ -1,16 +1,88 @@
...
+尽管 Linux/macOS 系统已经内置了 Python 解释器，但是建议用户不要使用它，
```
Comment (core-man) [inline source/programming/python.md:17]:
```suggestion
尽管 Linux/macOS 系统内置了 Python 解释器，但是建议用户不要使用它，
```

File: source/programming/python.md [hunk 14]
```diff
@@ -1,16 +1,88 @@
...
+以免误操作破坏系统内置 Python 造成系统出现问题。
```
Comment (core-man) [inline source/programming/python.md:18]:
```suggestion
以免误操作破坏系统内置 Python， 造成系统出现问题。
```

File: source/programming/python.md [hunk 15]
```diff
@@ -1,16 +1,88 @@
...
+**建议用户安装 Miniconda 并使用其提供的 `conda` 命令管理和安装 Python 及其模块。**
```
Comment (core-man) [inline source/programming/python.md:19]:
```suggestion
**建议用户安装 Miniconda， 使用其提供的 `conda` 命令管理和安装 Python 及其模块。**
```

File: source/programming/python.md [hunk 16]
```diff
@@ -1,16 +1,88 @@
...
+:color: info
+:icon: info
+
+在安装和使用 Miniconda 前，读者有必要了解 Python、Anaconda 和 Miniconda 三者
+之间的区别与联系：
+
+Python 解释器
+: 从 [Python 官方网站](https://www.python.org/downloads/) 下载的 Python 安装包
+  只提供了一个 Python 解释器，仅包含 Python 的核心模块和库，是运行 Python
+  脚本所必需的。
```
Comment (core-man) [inline source/programming/python.md:31]:
```suggestion
  脚本所必需的。安装 Python 解释器相当于安装了 Python 解释器 + 核心模块/库。
```

File: source/programming/python.md [hunk 17]
```diff
@@ -1,16 +1,88 @@
...
+
+Anaconda
+: [Anaconda](https://www.anaconda.com/) 是一个 Python **发行版**，不仅提供了
+  Python 解释器，还内置了很多 Python 开发工具与众多科学计算相关的库，形成了
+  一个可以开箱即用的 Python 科学计算环境，省去了自行配置科学计算环境的麻烦。
+  Anaconda 还提供了强大的软件包管理工具 `conda`，可以方便地安装模块和管理环境。
+  安装 Anaconda 相当于安装了 Python 解释器 + 数百个科学计算相关模块 + 包管理器 `conda`。
```
Comment (core-man) [inline source/programming/python.md:38]:
```suggestion
  安装 Anaconda 相当于安装了 Python 解释器 + 核心模块/库 + 数百个科学计算相关模块 + 包管理器 `conda`。
```

**Review by core-man — APPROVED** [review]

File: source/programming/python.md [hunk 18]
```diff
@@ -1,16 +1,88 @@
...
+功能强大等优点，是目前地震学科研最常用的编程语言之一。
+
+## 安装 Miniconda
+
+Python 是一种解释型语言，需要专门的解释器去执行 Python 代码。
+尽管 Linux/macOS 系统内置了 Python 解释器，但是建议用户不要使用它，
+以免误操作破坏系统内置 Python， 造成系统出现问题。
+**建议用户安装 Miniconda， 使用其提供的 `conda` 命令管理和安装 Python 及其模块。**
+
+:::{dropdown} Python vs Anaconda vs Miniconda
```
Comment (seisman) [inline source/programming/python.md:21]:
```suggestion
:::{dropdown} Python、Anaconda 和 Miniconda 的区别与联系
```

File: source/programming/python.md [hunk 19]
```diff
@@ -1,16 +1,88 @@
...
+  - 安装了很多平时用不到的模块，进而导致安装新模块时会可能出现版本冲突
+
+Miniconda
+: [Miniconda](https://docs.conda.io/en/latest/miniconda.html) 是 Anaconda 的精简版。
+  它继承了 Anaconda 的优点，同时避免了 Anaconda 的臃肿。其安装包只有约 50 MB，安装
+  通常也只需要数十秒。安装 Miniconda 相当于安装了 Python 解释器 + 核心模块/库 + 包管理器 `conda`。
 :::
 
-## Python 简介
+下面展示了如何在 Linux 系统下安装 Miniconda：
```
Comment (seisman) [inline source/programming/python.md:52]:
```suggestion
下面展示了如何在 Linux 系统下安装 Miniconda。其它操作系统下的安装说明以及具体使用方法可以参考 {doc}`Anaconda <software:anaconda/index>`。
```

File: source/programming/python.md [hunk 20]
```diff
@@ -1,16 +1,88 @@
...
+
+   Miniconda 默认会安装到 {file}`${HOME}/miniconda3` 下，在安装过程中可以
+   设置为其他路径。
+
+   安装通常只需要十几秒，在安装的最后会出现：
+
+        Do you wish the installer to initialize Miniconda3 by running conda init? [yes|no]
+        [yes] >>>
+
+   建议输入 `yes`，此时安装包会向当前 SHELL 的配置文件写入 `conda` 初始化语句。
```
Comment (seisman) [inline source/programming/python.md:72]:
```suggestion
   输入 `yes` 则安装包会向当前 SHELL 的配置文件写入 `conda` 初始化语句。
```

File: source/programming/python.md [hunk 21]
```diff
@@ -1,16 +1,88 @@
...
+        $ python
+        Python 3.9.12 (main, Apr  5 2022, 01:53:17)
+        [Clang 12.0.0 ] :: Anaconda, Inc. on darwin
+        Type "help", "copyright", "credits" or "license" for more information.
+        >>>
 
-Linux/macOS 都内置了 Python，足够日常使用，但强烈建议不要使用系统内置的 Python，
-而建议安装 [Anaconda](https://www.anaconda.com/products/individual)，并使用
-`conda` 管理 Python 环境。详情请参考 {doc}`Anaconda <software:anaconda/index>`。
+   在提示符 `>>>` 后输入 `quit()` 后按下 {kbd}`Enter` 键退出 Python。
```
Comment (seisman) [inline source/programming/python.md:85]:
```suggestion
   在提示符 `>>>` 后输入 `quit()` 后按下 {kbd}`Enter` 键退出 Python 解释器。
```

File: source/programming/python.md [hunk 22]
```diff
@@ -1,16 +1,88 @@
...
+
+        Do you wish the installer to initialize Miniconda3 by running conda init? [yes|no]
+        [yes] >>>
+
+   建议输入 `yes`，此时安装包会向当前 SHELL 的配置文件写入 `conda` 初始化语句。
+
+3. 测试安装
+
+   打开一个新的终端，在终端中输入 `python`，输出中看到 **Anaconda, Inc.**
+   字样即代表安装完成：
```
Comment (seisman) [inline source/programming/python.md:77]:
```suggestion
   字样即代表成功安装 Miniconda 并启动了 Python 解释器：
```

File: source/programming/python.md [hunk 23]
```diff
@@ -1,16 +1,88 @@
...
+   $ bash Miniconda3-latest-Linux-x86_64.sh
+   ```
+
+   Miniconda 默认会安装到 {file}`${HOME}/miniconda3` 下，在安装过程中可以
+   设置为其他路径。
+
+   安装通常只需要十几秒，在安装的最后会出现：
+
+        Do you wish the installer to initialize Miniconda3 by running conda init? [yes|no]
+        [yes] >>>
```
Comment (seisman) [inline source/programming/python.md:70]:
```suggestion
        Do you wish the installer to initialize Miniconda3 
        by running conda init? [yes|no]
        [no] >>>
```

File: source/programming/python.md [hunk 24]
```diff
@@ -1,16 +1,88 @@
...
+
+3. 测试安装
+
+   打开一个新的终端，在终端中输入 `python`，输出中看到 **Anaconda, Inc.**
+   字样即代表成功安装 Miniconda 并启动了 Python 解释器：
 
-## Python 与 Anaconda
+        $ python
+        Python 3.9.12 (main, Apr  5 2022, 01:53:17)
+        [Clang 12.0.0 ] :: Anaconda, Inc. on darwin
```
Comment (seisman) [inline source/programming/python.md:81]:
```suggestion
        [GCC 7.5.0] :: Anaconda, Inc. on linux
```

**Pull request merged**